### PR TITLE
[Renaming] Apply rename fully qualified namespace docblock on RenameNamespaceRector

### LINKED ISSUE #7045: Rename namespace rector doesn't update absolute references in PHPDoc's
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | 0.12.17             |

The rule `Rector\Renaming\Rector\Namespace_\RenameNamespaceRector`  doesn't update absolute class references.

For example:

I want rename namespace App\Entity to App\Entity\OrmEntity.

Running the below rector rule:

```
<?php

declare(strict_types=1);

use Rector\Core\Configuration\Option;
use Rector\Php74\Rector\Property\TypedPropertyRector;
use Rector\Renaming\Rector\Namespace_\RenameNamespaceRector;
use Rector\Set\ValueObject\LevelSetList;
use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;

return static function (ContainerConfigurator $containerConfigurator): void {
    // get parameters
    $parameters = $containerConfigurator->parameters();
    $parameters->set(Option::PATHS, [
        __DIR__.'/src',
        __DIR__.'/tests',
    ]);

    // Define what rule sets will be applied
//    $containerConfigurator->import(LevelSetList::UP_TO_PHP_81);

    // get services (needed for register a single rule)
    $services = $containerConfigurator->services();

    // register a single rule
    // $services->set(TypedPropertyRector::class);

    $services->set(RenameNamespaceRector::class)
        ->configure([
            'App\\Entity' => 'App\\Entity\\OrmEntity',
        ]);
};

```

Will not update references which are absolute in PHPDocs.

For example class \App\Entity\Book will not be renamed to \App\Entity\OrmEntity\Book.

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/7045

### CODE REVIEW

**Comment by samsonasik** [comment]
Test error is unrelated:

```bash
 There was 1 error:

1) Rector\Core\Tests\NonPhpFile\Rector\RenameClassNonPhpRector\RenameClassNonPhpRectorTest::test with data set #12 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
Error: Call to undefined method Nette\Neon\Node\BlockArrayNode::getSubNodes()

```

It seems due to new neon release https://github.com/nette/neon/releases/tag/v3.3.3

**Comment by samsonasik** [comment]
I temporary pin nette/neon to 3.3.2 https://github.com/rectorphp/rector-src/pull/1917/commits/fc57c0c892e036e85057e6875ef964f9eabb8f3d due to it can't be just update dependency and change loop usage, as it will cause error in rector-nette tests, ref https://github.com/rectorphp/rector-nette/pull/59

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/Renaming/Rector/Namespace_/RenameNamespaceRector.php [hunk 1]
```diff
@@ -124,6 +163,75 @@ public function configure(array $configuration): void
...
+            } else {
+                $value->type = $newType;
+            }
+        }
+
+        if ($value->type instanceof BracketsAwareUnionTypeNode) {
+            $value->type->types = $types;
+        }
+    }
+
```
Comment (TomasVotruba) [inline rules/Renaming/Rector/Namespace_/RenameNamespaceRector.php]: Can you extract this logic to external service? The rule is  getting too complex

Comment (samsonasik) [reply rules/Renaming/Rector/Namespace_/RenameNamespaceRector.php]: implemented, I created `DocBlockNamespaceRenamer` service for it https://github.com/rectorphp/rector-src/pull/1917/commits/17838a83b04ca1eb2f947dc8a45ea957cefd4c68

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**